cmd/nerdctl/build_nix_linux.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/containerd/continuity/fs"
+	"github.com/containerd/nerdctl/pkg/defaults"
+	"github.com/containerd/nerdctl/pkg/idgen"
+	"github.com/containerd/nerdctl/pkg/nixutil"
+	"github.com/opencontainers/go-digest"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func newBuildNixCommand() *cobra.Command {
+	shortHelp := "Reproducible image building with Nix derivations (EXPERIMENTAL)"
+	longHelp := shortHelp + `
+Examples: https://github.com/containerd/nerdctl/tree/master/examples/nix-build
+
+WARNING: This command is experimental and its behavior is subject to change.
+`
+	cmd := &cobra.Command{
+		Use:           "build-nix",
+		Short:         shortHelp,
+		Long:          longHelp,
+		RunE:          buildNixAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Args:          cobra.NoArgs,
+	}
+	cmd.Flags().String("nix-image", defaults.NixImage, "Nix image")
+	cmd.Flags().StringP("file", "f", "default.nix", "Name of the Nix derivation")
+	cmd.Flags().String("iidfile", "", "Write the image ID to the file")
+	return cmd
+}
+
+func buildNixAction(cmd *cobra.Command, args []string) error {
+	logrus.Warn("This command is experimental and subject to change.")
+
+	nixImage, err := cmd.Flags().GetString("nix-image")
+	if err != nil {
+		return err
+	}
+
+	file, err := cmd.Flags().GetString("file")
+	if err != nil {
+		return err
+	}
+	if _, err := os.Stat(file); err != nil {
+		return err
+	}
+	dir := filepath.Dir(file)
+	dir, err = filepath.Abs(dir)
+	if err != nil {
+		return err
+	}
+
+	iidfile, err := cmd.Flags().GetString("iidfile")
+	if err != nil {
+		return err
+	}
+
+	volStore, err := getVolumeStore(cmd)
+	if err != nil {
+		return err
+	}
+	if _, err := volStore.Get(nixutil.CacheVolumeName); err != nil {
+		if !errors.Is(err, os.ErrNotExist) && !errdefs.IsNotFound(err) {
+			return err
+		}
+		logrus.Debugf("Creating the cache volume %q", nixutil.CacheVolumeName)
+		if _, err := volStore.Create(nixutil.CacheVolumeName, nil); err != nil {
+			return err
+		}
+	} else {
+		logrus.Debugf("Using the existing cache volume %q", nixutil.CacheVolumeName)
+	}
+	tmpVolName := idgen.GenerateID()
+	logrus.Debugf("Creating a tmp volume %q for storing the Nix derivation file directory %q", tmpVolName, dir)
+	tmpVol, err := volStore.Create(tmpVolName, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		logrus.Debugf("Removing tmp volume %q", tmpVolName)
+		if _, rmVolErr := volStore.Remove([]string{tmpVolName}); rmVolErr != nil {
+			logrus.WithError(rmVolErr).Warnf("Failed to remove volume %q", tmpVolName)
+		}
+	}()
+	if err := fs.CopyDir(tmpVol.Mountpoint, dir); err != nil {
+		return err
+	}
+
+	logrus.WithField("image", nixImage).Infof("Creating a Nix container")
+	nerdctlCmd, nerdctlArgs := globalFlags(cmd)
+	runCmd := exec.Command(nerdctlCmd, append(nerdctlArgs, []string{
+		"run",
+		"-d",
+		"-v", tmpVolName + ":/mnt",
+		"-v", nixutil.CacheVolumeName + ":/nix",
+		"-w", "/mnt",
+		nixImage,
+		"sleep", "infinity",
```

Comment:
I assume that a non-null   entrypoint can cause problems

Reply:
why?

Reply:
`cmd.Flags().String("nix-image", defaults.NixImage, "Nix image")`  
`nix-image` can be set by the user and nothing prevents having an image with an entrypoint ["/ bin/sh /," sleep","infinity"] => `/bin/sh sleep infinity sleep infinity`. 


I know if this should be considered the responsibility of the nerdctl user or us